daemon/cmd/daemon_main.go

```diff
@@ -730,10 +730,6 @@ func InitGlobalFlags(cmd *cobra.Command, vp *viper.Viper) {
 	flags.Bool(option.Restore, true, "Restores state, if possible, from previous daemon")
 	option.BindEnv(vp, option.Restore)
 
-	flags.String(option.SidecarIstioProxyImage, k8s.DefaultSidecarIstioProxyImageRegexp,
```

Comment:
i think we should mention the removed flags and helm values in the upgrade guide. (in addition to https://github.com/cilium/cilium/pull/25722)

Reply:
ah good point :+1: 

Let's see if there is other comments, which requires for code change. Otherwise, I will update docs in separate PR, just to avoid CI  re-run.

Reply:
Done as part of rebase to solve conflict.

Reply:
what about mentioning the remove flag `--sidecar-istio-proxy-image` too? (in case someone is using is using Cilium without the Helm Chart)